libheif/heif_items.cc

```diff
@@ -32,30 +32,30 @@
 
 // ------------------------- reading -------------------------
 
-int heif_context_get_number_of_items(const struct heif_context* ctx)
+size_t heif_context_get_number_of_items(const struct heif_context* ctx)
```

Comment:
I'd like to avoid `size_t` as much as possible in the interface because that one bit of extra range is not needed here and `size_t` makes detecting errors much more complicated:
[See Scott Meyer](http://www.aristeia.com/Papers/C++ReportColumns/sep95.pdf)

I'm probably fine with `size_t` when specifying the size of memory areas that may be so large to hit the 2GB boundary on 32bit systems.

Reply:
I'm not sure I agree with the analysis. Even in the quoted article, there are plenty of other cases where signed int won't save you (e.g. the off-by-one case).

Internally we're just casting this to `size_t` anyway, and a common case is to pass in the `std::vector` size, so I'm casting to `int` to match the API, then its being cast back...

Possibly we can use `size_t` and then just check the number isn't greater than some limit, similar to the way we do security checks.

Reply:
To get this moving, I've removed the `size_t` changes.

The unit test still fails.